influxdb3_write/src/lib.rs

```diff
@@ -240,6 +244,7 @@ impl PersistedSnapshot {
         catalog_sequence_number: SequenceNumber,
     ) -> Self {
         Self {
+            last_file_id: NEXT_FILE_ID.load(Ordering::SeqCst),
```

Comment:
If you're setting this on load, I think you should also be setting it to whatever the max is when you `add_parquet_file` gets called.

Reply:
when calling `add_parquet_file` we call `fetch_add` this will [return the current value and increment by 1 for other threads to see](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html#method.fetch_add). So `NEXT_FILE_ID` *is* the next unique file id. If we use the last one from our call to `add_parquet_file` we would be reusing an id

Reply:
It might be more clear if I label it `next_unqiue_file_id` instead

Reply:
Not quite what I'm suggesting. Load here is correct. However, if you create this persisted snapshot struct, and then you create some parquet files, which increment the `NEXT_FILE_ID` value, it won't be picked up by the snapshot, unless when you call `add_parquet_file` on the snapshot, you set `last_file_id` like `self.last_file_id = self.last_file_id.max(parquet_file.id)`

I didn't see that happening in there, so you could have a stale `last_file_id` value.

Reply:
OH I SEE NOW. We create a persisted snapshot type that uses the value 500 f.ex and if we add a file we now increase the value by one and so the next value to use should be 501 in this example not the 500 that's persisted.

Reply:
Fixed this issue so that the largest file id added to the PersistedSnapshot is then increased by 1 as the next available id.